### PR TITLE
Remove Collectable; rely on collection builders instead.

### PR DESCRIPTION
**Pull request by greg-1-anderson**
The collection builder API is better; removing Collectable eliminates a lot of redundant methods from BaseTask.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,15 +3,15 @@
...
-   * Tasks may also be added to a collection via `$collection->add($task);`
+   * Tasks may be added to a collection via `$collection->add($task);`
    * `$collection->run();` runs all tasks in the collection
    * `$collection->addCode(function () { ... } );` to add arbitrary code to a collection
    * `$collection->progressMessage(...);` will log a message
    * `$collection->rollback($task);` and `$collection->rollbackCode($callable);` add a rollback function to clean up after a failed task
    * `$collection->completion($task);` and `$collection->completionCode($callable);` add a function that is called once the collection completes or rolls back.
    * `$collection->before();` and `$collection->after();` can be used to add a task or function that runs before or after (respectively) the specified named task. To use this feature, tasks must be given names via an optional `$taskName` parameter when they are added.
    * Collections may be added to collections, if desired. 
+* [CollectionBuilder] Create tasks and add them to a collection in a single operation.
```
Comment (DavertMik) [inline CHANGELOG.md]: should include an example even in changelog

File: CHANGELOG.md [hunk 2]
```diff
@@ -22,7 +22,7 @@
    * *Breaking* Tasks that use other tasks must also use `$this->task('taskName');` instead of `new TaskClass();` when creating task objects to call.
 * [Extract] task added
 * [Pack] task added
-* [TmpFile] task added
+* [TmpDir], [WorkDir] and [TmpFile] tasks added
 * Support standalone Robo scripts that allows scripts starting with `#!/usr/bin/env robo` to define multiple robo commands.  Use `#!/usr/bin/env robo run` to define a single robo command implemented by the `run()` method.
 * Provide ProgresIndicatorAwareInterface and ProgressIndicatorAwareTrait that make it easy to add progress indicators to tasks
 * Add --simulate mode that causes tasks to print what they would have done, but make no changed
```
Comment (DavertMik) [inline CHANGELOG.md]: no change**s**

File: docs/collections.md [hunk 3]
```diff
@@ -119,37 +119,53 @@ Since the concept of temporary objects that are cleaned up  on failure is a comm
...
+class RoboFile extends \Robo\Tasks
+{
+    function myOperation()
+    {
+        $collection = $this->collectionBuilder();
+        
+        // Create a temporary directory, and fetch its path.
+        // If all of the tasks succeed, then rename the temporary directory
+        // to its final name.
+        $work = $collection->workDir('destination');
```
Comment (DavertMik) [inline docs/collections.md]:
```
$collection->workDir('destination')
```
is that a real method?

could you provide real tasks in examples?

File: docs/collections.md [hunk 4]
```diff
@@ -119,37 +119,53 @@ Since the concept of temporary objects that are cleaned up  on failure is a comm
...
+
+        $collection->taskOther($work);
```
Comment (DavertMik) [inline docs/collections.md]: PLease add real task here too

File: docs/collections.md (hunk 3 above)
Comment (greg-1-anderson) [inline docs/collections.md]: workDir is a real method.  I'll find a real task for here and below.

**Comment by greg-1-anderson** [comment]
Updated, squashed and force-pushed. @DavertMik LMK if this looks ready to merge now.

File: CHANGELOG.md (hunk 1 above)
Comment (greg-1-anderson) [inline CHANGELOG.md]: GitHub isn't showing it, but there's an example on the next line now.

**Comment by DavertMik** [comment]
coool!

**Pull request merged**